src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
-/// `(label, (point_label, point))`, where `label` is the label of a polynomial in `p`,
-/// `point_label` is the label for the point (e.g., "beta"), and  and `point` is the field element
-/// that `p[label]` is to be queried at.
+/// `QuerySet` is the set of queries that are to be made to a set of labeled polynomials or linear combinations.
+///
+///  Each element of a `QuerySet` is a pair of `(label, (point_label, point))`, where
+///  * `label` is the label of a polynomial in `p`,
+///  * `point_label` is the label for the point (e.g., "beta"), and  and
+///  * `point` is the field element that `p[label]` is to be queried at.
 pub type QuerySet<'a, F> = BTreeSet<(String, (String, F))>;
 
-/// `Evaluations` is the result of querying a set of labeled polynomials or equations
-/// `p` at a `QuerySet` `Q`. It maps each element of `Q` to the resulting evaluation.
+/// `Evaluations` is the result of querying a set of labeled polynomials or linear combinations
+/// `p` at a `QuerySet` `Q`.
+///
+/// It maps each element of `Q` to the resulting evaluation.
 /// That is, if `(label, query)` is an element of `Q`, then `evaluation.get((label, query))`
 /// should equal `p[label].evaluate(query)`.
 pub type Evaluations<'a, F> = BTreeMap<(String, F), F>;
 
-/// A proof of satisfaction of linear combinations.
-#[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
-pub struct BatchLCProof<F: Field, PC: PolynomialCommitment<F>> {
-    /// Evaluation proof.
-    pub proof: PC::BatchProof,
-    /// Evaluations required to verify the proof.
-    pub evals: Option<Vec<F>>,
-}
-
-impl<F: Field, PC: PolynomialCommitment<F>> SemanticallyValid for BatchLCProof<F, PC> {
-    fn is_valid(&self) -> bool {
-        self.proof.is_valid()
-            && if self.evals.is_some() {
-                self.evals.as_ref().unwrap().is_valid()
-            } else {
-                true
-            }
-    }
-}
-
-/// Describes the interface for a polynomial commitment scheme that allows
-/// a sender to commit to multiple polynomials and later provide a succinct proof
-/// of evaluation for the corresponding commitments at a query set `Q`, while
-/// enforcing per-polynomial degree bounds.
-pub trait PolynomialCommitment<F: Field>: Sized {
-    /// The universal parameters for the commitment scheme. These are "trimmed"
-    /// down to `Self::CommitterKey` and `Self::VerifierKey` by `Self::trim`.
-    type UniversalParams: PCUniversalParams;
+/// Describes the interface for a homomorphic polynomial commitment scheme with values
+/// in a commitment group `G`.  
+///
+/// The simple single-polynomial evaluation proof depends on the conrete scheme, whereas
+/// more expressive opening proofs are reduced to a simple one by leveraging the homomorphic
+/// properties.
+/// The interface comes with the additional feature of splitting the verifier into a
+/// `succinct` part and `non-succinct` part.   
+pub trait PolynomialCommitment<G: Curve>: Sized {
```

Comment:
We should think about circuit friendliness too.
I believe this trait should be templatized with G: EndoMulCurve and not simply curve.
This was already done in ginger-lib as part of:

https://github.com/HorizenOfficial/ginger-lib/pull/167